src/gui/qgsmaptoolzoom.h

```diff
@@ -58,6 +58,9 @@ class GUI_EXPORT QgsMapToolZoom : public QgsMapTool
     //! Flag to indicate a map canvas drag operation is taking place
     bool mDragging;
 
+    //! Flag to indicate the user has canceled the current zoom operation
+    bool mCanceled;
```

Comment:
```suggestion
    bool mCanceled = false;
```

As mentioned by Nyall, you must initialize it at false.